doc/sphinxext/allow_nan.py

```diff
@@ -0,0 +1,98 @@
+from sklearn.utils import all_estimators
+from sklearn.linear_model import LogisticRegression
+from sklearn.preprocessing import OrdinalEncoder, OneHotEncoder
+from sklearn.compose import ColumnTransformer
+from io import StringIO
+from docutils import nodes
+
+from docutils.parsers.rst import Directive
+
+
+class Allow_Nan(Directive):
+    def run(self):
+        names1 = []
+        names2 = []
+        names3 = []
+        names4 = []
+        output = StringIO()
+        output.write(
+            f"List of estimators that allow NaN values for type_filter=cluster : "
```

Comment:
From the [rendering](https://168955-843222-gh.circle-artifacts.com/0/doc/modules/impute.html#estimators-that-handle-nan-values) this first list is empty. Would it be possible to avoid printing the list when the list itself is empty?